lib/grid-register.js

```diff
@@ -55,6 +55,13 @@ async function postRequest (data, addr, port) {
     // re-serialize the configuration with the auto populated data
     data = JSON.stringify(jsonObject);
   }
+  // if the node config does not have id automatically add it
+  if (!jsonObject.configuration.id) {
+    jsonObject.configuration.id = `http://${jsonObject.configuration.host}:${jsonObject.configuration.port}`;
+
+    // re-serialize the configuration with the auto populated data
+    data = JSON.stringify(jsonObject);
```

Comment:
This, and the one above, should be moved out and only done once, so we end up with

```js
if (!jsonObject.configuration.url || !jsonObject.configuration.host || !jsonObject.configuration.port) {
  jsonObject.configuration.url = `http://${addr}:${port}/wd/hub`;
  jsonObject.configuration.host = addr;
  jsonObject.configuration.port = port;
}
// if the node config does not have id automatically add it
if (!jsonObject.configuration.id) {
  jsonObject.configuration.id = `http://${jsonObject.configuration.host}:${jsonObject.configuration.port}`;
}

// re-serialize the configuration with the populated data
data = JSON.stringify(jsonObject);
```

There is little overhead to stringifying the JSON object.